### PR TITLE
feat(core/id-headers): add section links

### LINKED ISSUE #1821: Add self-links to headings (§ hyperlink anchor)
It would be great if Respec-generated specs had self-links for all headings, the way the HTML spec does and that Bikeshed-generated specs do.

To be clear, I mean the feature where, in the left margin just before the heading, there’s a `§` character that’s a hyperlink to the URL for that heading.

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #1821

### CODE REVIEW

File: src/core/id-headers.js [hunk 1]
```diff
@@ -3,13 +3,32 @@
...
+      addId(elem);
+      if (conf.addSectionLinks) {
+        addSectionLink(elem);
+      }
     });
 }
+
+function addSectionLink(h) {
+  const section = h.closest("section[id]");
+  const a = document.createElement("a");
```
Comment (sidvishnoi) [inline src/core/id-headers.js]: We could use `hyperHTML` here maybe

File: src/core/id-headers.js [hunk 2]
```diff
@@ -3,13 +3,32 @@
...
+  a.classList.add("self-link");
+  const id = section ? section.id : h.id;
+  a.href = `#${id}`;
+  a.setAttribute("aria-label", "§");
+  h.insertAdjacentElement("beforeend", a);
```
Comment (sidvishnoi) [inline src/core/id-headers.js]: Why not `h.apppendChild(a)`?

File: tests/spec/core/id-headers-spec.js [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
+    it("(aria) labels section links", () => {
+      const queryForSymbol = "h2 a[aria-label=§].self-link";
+      expect(doc.querySelector(queryForSymbol)).toBeTruthy();
+    });
+
+    it("doesn't add sections links when addSectionLinks is false", async () => {
+      const conf = {
+        addSectionLinks: false,
+      };
+      const ops = makeStandardOps(conf, body);
```
Comment (sidvishnoi) [inline tests/spec/core/id-headers-spec.js]: Nit:
``` js
makeStandardOps({ addSectionLinks: false }, body);
```

File: js/core/css/respec2.css [hunk 4]
```diff
@@ -171,6 +171,64 @@ details.respec-tests-details>li {
...
+
+h2,
+h3,
+h4,
+h5,
+h6 {
+  position: relative;
+}
+
+h2>a.self-link,
```
Comment (sidvishnoi) [inline js/core/css/respec2.css]: Do we really need this level of specificity?

File: src/core/id-headers.js (hunk 2 above)
Comment (sidvishnoi) [reply src/core/id-headers.js]:
``` js
function addSectionLink(h) {
  const section = h.closest("section[id]");
  const id = section ? section.id : h.id;
  h.appendChild(hyperHTML`
    <a href="${'#' + id}" class="self-link" aria-label="§"></a>
  `);
}
```

Comment (marcoscaceres) [reply src/core/id-headers.js]: I want it at the end tho :)

Comment (marcoscaceres) [reply src/core/id-headers.js]: oh wait, that is at end... I should stop

File: js/core/css/respec2.css (hunk 4 above)
Comment (marcoscaceres) [reply js/core/css/respec2.css]: I think so, as we might want to add different self-link types.

File: tests/spec/core/id-headers-spec.js [hunk 5]
```diff
@@ -1,13 +1,50 @@
...
-    expect($s.attr("id")).toEqual("foo");
+  let doc;
+  const body = `
+    <section class="introductory"><h2>Intro</h2></section>
+    <section id="t0"><p>BLAH</p><h6>FOO</h6></section>
+    <section><h2>test-1</h2></section>
+    <section><h2>Pass</h2></section>
+  `;
+  beforeAll(async () => {
+    const ops = makeStandardOps({ addSectionLinks: false }, body);
```
Comment (sidvishnoi) [inline tests/spec/core/id-headers-spec.js]: `addSectionLinks: false` oops?

File: tests/spec/core/id-headers-spec.js [hunk 6]
```diff
@@ -29,10 +29,12 @@ describe("Core - ID headers", () => {
     });
 
     it("doesn't add sections links when addSectionLinks is false", async () => {
-      const conf = {
-        addSectionLinks: false,
-      };
-      const ops = makeStandardOps(conf, body);
+      const ops = makeStandardOps(
+        {
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/id-headers-spec.js]: heh, yeah

**Comment by marcoscaceres** [comment]
> I thought I would be working on this one to get back 

Got plenty of bugs for ya, don't worry 😄

**Comment by marcoscaceres** [comment]
documentation https://github.com/w3c/respec/wiki/addSectionLinks

**Pull request merged**